cavern/README.md

```diff
@@ -1,44 +1,100 @@
+Because the source of all data and metadata in `cavern` is the file system, users may interact with that file system 
+directly (through, for example, a volume mount) and `cavern` will reflect any changes that were made. 
 
-Because the source of all data and metadata in Cavern is the file system, users may interact with that file system directly (through, for example, a volume mount) and cavern will reflect any changes that were made. 
+## deployment
+The `cavern` war file can be renamed at deployment time in order to support an alternate service name, including 
+introducing additional path elements. 
+See <a href="https://github.com/opencadc/docker-base/tree/master/cadc-tomcat">cadc-tomcat</a> (war-rename.conf).
 
-## building
-```
-> gradle clean build
+## configuration
+The following runtime configuration must be made available via the `/config` directory.
 
-> docker build -t images.canfar.net/skaha-system/cavern:latest -f Dockerfile .
+### catalina.properties
+This file contains java system properties to configure the tomcat server and some of the java libraries 
+used in the service.
+
+See <a href="https://github.com/opencadc/docker-base/tree/master/cadc-tomcat">cadc-tomcat</a> for 
+system properties related to the deployment environment.
+
+See <a href="https://github.com/opencadc/core/tree/master/cadc-util">cadc-util</a> for common system properties.
+
+`cavern` includes multiple IdentityManager implementations to support authenticated access:
+- See <a href="https://github.com/opencadc/ac/tree/master/cadc-access-control-identity">cadc-access-control-identity</a> for CADC access-control system support.
+- See <a href="https://github.com/opencadc/ac/tree/master/cadc-gms">cadc-gms</a> for OIDC token support.
+
+`cavern` requires a connection pool to the local UWS database:
+```
+# database connection pools
+org.opencadc.cavern.uws.maxActive={max connections for jobs pool}
+org.opencadc.cavern.uws.username={database username for jobs pool}
+org.opencadc.cavern.uws.password={database password for jobs pool}
+org.opencadc.cavern.uws.url=jdbc:postgresql://{server}/{database}
 ```
-Then, update VERSION with the approripate tages and run ./apply-version.sh
 
-## running it
+The `uws` pool manages (create, alter, drop) uws tables and manages the uws content.
+
+### cadc-registry.properties
+See <a href="https://github.com/opencadc/reg/tree/master/cadc-registry">cadc-registry</a>.
+
+### cavern.properties
+A `cavern.properties` file in /config is required to run this service.  The following keys are required:
 ```
-docker run -d --volume=/path/to/external/config:/config:ro --name cavern-tomcat images.canfar.net/skaha-system/cavern:latest
+# service identity
+org.opencadc.cavern.resourceID = ivo://{authority}/{name}
+
+# base directory for cavern files
+org.opencadc.cavern.filesystem.root = {directory path}
+
```

Comment:
let's change the above to
```
org.opencadc.cavern.filesystem.baseDir = {persistent data directory in container}
org.opencadc.cavern.filesystem.subPath = {relative path to the node/file content that could be mounted in other containers}
```

then remove the probe config and hard-code the subdir; for now we can hard code the `tomcat` for those tests

Reply:
Updated the README, will update the tests.